### ISSUE TITLE
Slate transform does not roundtrip when the text uses softbreaks

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/markdown-transform#44]
**Describe the bug**
Translating from markdown to slate and back sometimes introduces additional new lines.

**To Reproduce**

For instance on the attached sample.md:
[sample.txt](https://github.com/accordproject/markdown-transform/files/3687219/sample.txt)

```sh
$ cat sample.md 
## Fixed rate loan

This is a _fixed interest_ loan to the amount of 100000
at the yearly interest rate of 2.5%
with a loan term of 15,
and monthly payments of {{I'm not sure which amount right now}}
$ markus normalize --sample sample.md --slate
12:34:44 - info: 
Fixed rate loan
----

This is a *fixed interest* loan to the amount of 100000

at the yearly interest rate of 2.5%

with a loan term of 15,

and monthly payments of {{I'm not sure which amount right now}}
```

**Comment by jeromesimeon** [comment]
This issue is most likely related to softbreaks which gets translated into paragraphs. See also https://github.com/accordproject/template-studio-v2/issues/147

**Comment by dselman** [comment]
I think you are correct: https://github.com/accordproject/markdown-transform/blob/master/packages/markdown-slate/src/ToSlateVisitor.js#L235

**Comment by jeromesimeon** [comment]
I'm not sure what is the right way to address this. A couple of possible directions:
1. Preserve softbreaks in the Slate DOM and delay the new line creation until rendering (this might be useful if we want to allow users to control that when editing)
2. Turn softbreaks into a new line folded in the enclosing paragraph (a bit like the HTML transform), but that means we need to recover those

As I'm writing it down, I think like option 1. better but it might be more work (since this is a new kind of Slate node to handle in the editor)... There might be other options to handle this that I'm not thinking about.

**Comment by dselman** [comment]
Yes, I think I like (1) better as well. E.g. is would allow CTRL-ENTER in the editor to insert a softbreak.

FYI, here is what the CommonMark spec says: https://spec.commonmark.org/0.22/#soft-line-breaks

**Comment by jeromesimeon** [comment]
> Yes, I think I like (1) better as well. E.g. is would allow CTRL-ENTER in the editor to insert a softbreak.
> 
> FYI, here is what the CommonMark spec says: https://spec.commonmark.org/0.22/#soft-line-breaks

Yes. That makes sense. @mttrbrts encountered this as well on the HTML side when he looked at pdf rendering, so he may have some thoughts?

**Comment by jeromesimeon** [comment]
This bug was fixed in #113 

```
bash-3.2$ markus normalize --sample sample.md --slate
15:02:01 - info: 
Fixed rate loan
----

This is a *fixed interest* loan to the amount of 100000

at the yearly interest rate of 2.5%

with a loan term of 15,

and monthly payments of {{I'm not sure which amount right now}}
bash-3.2$ ./packages/markdown-cli/index.js normalize --sample sample.md --slate
15:02:10 - info: 
Fixed rate loan
----

This is a *fixed interest* loan to the amount of 100000
at the yearly interest rate of 2.5%
with a loan term of 15,
and monthly payments of {{I'm not sure which amount right now}}
```
